internal/collector/collector.go

```diff
@@ -86,6 +99,16 @@ func GetConfig(id string) (Config, error) {
 	return config, nil
 }
 
+// createArchive compresses a directory into a xz-compressed tar archive.
+// Returns an error if the tar command fails.
+func createArchive(archiveName, tmpDir string) error {
+	cmd := exec.Command("tar", "-cJf", archiveName, "-C", tmpDir, ".")
+	if err := cmd.Run(); err != nil {
```

Comment:
Do we care about whatever `tar` may print to standard output and error? If so, `cmd.CombinedOutput()` or using `.StdinPipe()+.StderrPipe()` might provide ways to enure our log contains useful information.